File: 프로그래머스/연습문제/Lv2/할인 행사/할인 행사.py
def solution(want, number, discount):
    want_list = []
    for i in range(len(want)):
        for j in range(number[i]):
            want_list.append(want[i])
    want_list = sorted(want_list)
    
    cnt = 0
    for i in range(len(discount)-9):
        cmp_list = sorted(discount[i:i+10])
        if want_list == cmp_list: cnt += 1
        
    return cnt

File: 프로그래머스/연습문제/Lv2/할인 행사/할인 행사.cpp
#include <string>
#include <vector>
#include <map>
using namespace std;

int solution(vector<string> want, vector<int> number, vector<string> discount) {
    map<string, int> m; // <품목, 원하는 개수>
    for(int i=0; i<want.size(); i++)
        m[want[i]] = number[i];
    
    int cnt = 0;
    for(int i=0; i<=discount.size()-10; i++){
        map<string, int> cmp;
        for(int j=0; j<10; j++)
            cmp[discount[i+j]]++;
        
        if(m == cmp) cnt++;
    }
    
    return cnt;
}
